### PR TITLE
fix #39 Java11 build

### LINKED ISSUE #39: Make the release build work with jdk11
Currently, it is not possible to do a release build of jasper-jsp with jdk11, only jdk8. Need to make it work with jdk11.

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: Olivier Lamy <olamy@apache.org>

### CODE REVIEW

**Comment by olamy** [comment]
need to test/review manifest generation as upgrade necessary for java 17 but might break some stuff

File: pom.xml [hunk 1]
```diff
@@ -296,8 +245,7 @@
         <plugin>
           <groupId>org.apache.felix</groupId>
           <artifactId>maven-bundle-plugin</artifactId>
-          <!-- keep this version as upgrade break manifest -->
-          <version>4.0.0</version>
+          <version>5.1.4</version>
```
Comment (olamy) [inline pom.xml:248]: need to test/review manifest generation as upgrade necessary for java 17 but might break some stuff

**Review by joakime — APPROVED** [review]
Yes, please double-check the MANIFEST output here.
Otherwise it looks good.

**Comment by olamy** [comment]
obviously the produced jars/manifest doesn;t work anymore :)

**Comment by olamy** [comment]
@janbartel if you could have a look here. as now the generated MANIFEST doesn't work anymore with jetty-11 branch. should we custom this build to test the produced jars again the corresponding jetty branches?

**Comment by janbartel** [comment]
Why did you change to use the `jakarta.el:jakarta.el-api` jar instead of the `tomcat-el-api` jar, which the apache tomcat el implementation is known to work with? The code is different, if you compare the sources.

There are differences in the `apache-el` `MANIFEST.MF` generated by this branch compared to the `master` branch:

```
22,24c21,25
< Import-Package: jakarta.servlet.jsp;version="3.0";resolution:=optional,j
<  akarta.servlet.jsp.el;version="3.0";resolution:=optional,jakarta.servle
<  t.jsp.tagext;version="3.0";resolution:=optional
---
> Import-Package: jakarta.servlet.jsp;resolution:=optional,jakarta.servlet
>  .jsp.el;resolution:=optional,jakarta.servlet.jsp.tagext;resolution:=opt
>  ional
> Require-Capability: osgi.contract;osgi.contract=JavaJSP;filter:="(&(osgi
>  .contract=JavaJSP)(version=3.0.0))"
26c27
```

As you can see there are 2 problems here:
1. the version of the import statements has been lost
2. the `Require-Capability` statement has been added, but there is nothing in the `apache-jsp` manifest that provides that capability.

**Comment by olamy** [comment]
> Why did you change to use the `jakarta.el:jakarta.el-api` jar instead of the `tomcat-el-api` jar, which the apache tomcat el implementation is known to work with? The code is different, if you compare the sources.
> 

not sure why maybe wanted to use something more conventional as this should just be specs but yeah to many versions/distributions of the same specs :) 
Anyway reverting this doesn't change much.

> There are differences in the `apache-el` `MANIFEST.MF` generated by this branch compared to the `master` branch:
> 
> ```
> 22,24c21,25
> < Import-Package: jakarta.servlet.jsp;version="3.0";resolution:=optional,j
> <  akarta.servlet.jsp.el;version="3.0";resolution:=optional,jakarta.servle
> <  t.jsp.tagext;version="3.0";resolution:=optional
> ---
> > Import-Package: jakarta.servlet.jsp;resolution:=optional,jakarta.servlet
> >  .jsp.el;resolution:=optional,jakarta.servlet.jsp.tagext;resolution:=opt
> >  ional
> > Require-Capability: osgi.contract;osgi.contract=JavaJSP;filter:="(&(osgi
> >  .contract=JavaJSP)(version=3.0.0))"
> 26c27
> ```
> 
> As you can see there are 2 problems here:
> 
> 1. the version of the import statements has been lost
> 2. the `Require-Capability` statement has been added, but there is nothing in the `apache-jsp` manifest that provides that capability.

please note this upgrade to be able to build with jdk 17 as well `Created-By: Apache Maven Bundle Plugin 5.1.4` 
this guy is the cause of this change.

**Comment by olamy** [comment]
@janbartel I can remove this `maven-bundle-plugin` upgrade as it's only needed for java 17 build (not sure when Tomcat will have this as a minimum) so that's the ezy option :)

**Comment by joakime** [comment]
@olamy @janbartel is this still needed?

**Comment by olamy** [comment]
> @olamy @janbartel is this still needed?

I'm not quite sure... 
it looks more to fix jdk17 build now rather than 11 :)

**Comment by joakime** [comment]
@olamy the builds on this project need some love.
Most PRs are failing now.

**Pull request closed without merging**